src/Control/Effect/Internal/TH/Effect.hs

```diff
@@ -0,0 +1,145 @@
+{-# OPTIONS_HADDOCK not-home #-}
+
+-- This file is copied almost verbatim from polysemy.
+
+module Control.Effect.Internal.TH.Effect
+  ( makeEff,
+    makeEff_,
+  )
+where
+
+import Control.Effect.Internal.TH.Common
+  ( ConLiftInfo (..),
+    checkExtensions,
+    foldArrowTs,
+    getEffectMetadata,
+    makeMemberConstraint,
+    makeUnambiguousSend,
+  )
+import Control.Monad
+import Language.Haskell.TH
+import Language.Haskell.TH.Datatype
+
+-- TODO: write tests for what should (not) compile
```

Comment:
If Polysemy has such tests, I'd appreciate a port of them. Otherwise it's not critical.